### PR TITLE
Add endpoint to return publication info

### PR DESCRIPTION
**Pull request by dglemos**
Add endpoint:
- `gene2phenotype/api/publication/<pmid>`

[G2P-124](https://www.ebi.ac.uk/panda/jira/browse/G2P-124)

### CODE REVIEW

File: gene2phenotype_project/gene2phenotype_app/views.py [hunk 1]
```diff
@@ -259,6 +262,76 @@ def list(self, request, *args, **kwargs):
...
+                    title = publication_info['title']
+                    if 'pubYear' in publication_info:
+                        year = publication_info['pubYear']
+
+                    data.append({
+                        'pmid': int(pmid),
+                        'title': title,
+                        'authors': authors,
+                        'year': int(year),
+                        'source': 'EuropePMC (PMID not found in G2P)'
```
Comment (seeta-ramaraju) [inline gene2phenotype_project/gene2phenotype_app/views.py:327]:
```suggestion
                        'source': 'EuropePMC'
```

File: gene2phenotype_project/gene2phenotype_app/views.py [hunk 2]
```diff
@@ -259,6 +262,76 @@ def list(self, request, *args, **kwargs):
...
+    Return:
+            Response object includes:
+                - results (list): contains publication data for each publication
+                                    - pmid
+                                    - title
+                                    - authors
+                                    - year
+                                    - source (possible values: 'G2P', 'EuropePMC (PMID not found in G2P)', 'Not found')
+                - count (int): number of PMIDs in the response
+            If a PMID is invalid it returns Http404
```
Comment (seeta-ramaraju) [inline gene2phenotype_project/gene2phenotype_app/views.py:283]: Does it return 404 error?

Comment (dglemos) [reply gene2phenotype_project/gene2phenotype_app/views.py:283]: Yes

**Review by seeta-ramaraju — APPROVED** [review]
Looks good to me

**Pull request merged**